compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -603,6 +603,13 @@ impl<'a, Ty> FnAbi<'a, Ty> {
         Ty: TyAndLayoutMethods<'a, C> + Copy,
         C: LayoutOf<Ty = Ty, TyAndLayout = TyAndLayout<'a, Ty>> + HasDataLayout + HasTargetSpec,
     {
+        if abi == spec::abi::Abi::X86Interrupt {
```

Comment:
Could this change be extracted into a separate commit with a comment as to why it was made? This is also something that seems like it could explode if rustc was built with system llvm <=12.0.

Reply:
I've split this out into a separate commit with a reference to https://github.com/llvm/llvm-project/commit/2e0e03c6a089da39039ec3f464f7cee5df86646b. As far as I know, marking the argument byval on earlier LLVM versions is fine, just not necessary.

I've also adjusted the code to only mark the first argument byval. After looking a bit closer, it turns out that this ABI can have a second argument, and the second one shouldn't be byval.